benches/search_comparison.rs

```diff
@@ -285,3 +255,50 @@ fn make_sorted_vec<T: Ord>(mut v: Vec<T>) -> Vec<T> {
 fn search_sorted_vec<'a, T: Ord>(c: &'a Vec<T>, x: T) -> Option<&'a T> {
     c.binary_search(&x).ok().map(|i| &c[i])
 }
+
+/// Generate pseudorandom sequence of numbers
+///
+/// ```rust
+/// assert_eq!(
+///     pseudorandom_iter::<u32>(0, u32::MAX , None).take(16).collect(),
+///     vec![2027808446, 2393657406, 900912232, 833811770, 1061328792, 93432844,
+///     2565420364, 1550801266, 1147887774, 710446582, 3306204668, 500014398,
+///     1140212266, 2163551532, 513205252, 1774545590]
+///
+/// )
+/// ```
+///
+/// ```rust
+/// assert_eq!(
+///     pseudorandom_iter::<u8>(0, u8::MAX, None).take(16).collect(),
+///     vec![250, 200, 36, 20, 178, 78, 4, 152, 174, 8, 128, 198, 166, 206, 156, 80]
+/// );
+/// ```
+///
+/// ```rust
+/// assert_eq!(
+///     pseudorandom_iter::<u8>(0, u8::MAX, Some(32)).take(16).collect(),
+///     vec![27, 9, 5, 20, 18, 15, 5, 24, 15, 8, 1, 6, 7, 15, 29, 17]
+/// );
+fn pseudorandom_iter<T>(
+    mut seed: usize,
+    max: usize,
+    bound: Option<usize>,
```

Comment:
I'm pretty confused by the name of this, because it's not _just_ a bound, it also changes whether even/odd numbers are generated. I think we should either give it a name that dictates the use (`for_construction`?), or we should split `bound` and `even_only` like you originally suggested. I know I said getting the nuances of internal code right isn't that important (and I still believe that), but as written I just find this function confusing, which makes it more likely for bugs to sneak in :sweat_smile:

Reply:
Removed `bound` parameter